### ISSUE TITLE
HBA bridge unsatisfiable classes

### ISSUE THREAD

**Issue by hkir-dev** [obophenotype/uberon#2903]
When `uberon`, `uberon-bridge-to-hba` and `hba` ontologies merged and reasoned,  203 unsatisfiable classes are identified. 

```
robot merge -I http://purl.obolibrary.org/obo/uberon/uberon.owl -I https://raw.githubusercontent.com/obophenotype/uberon/master/src/ontology/bridge/uberon-bridge-to-hba.obo -I https://raw.githubusercontent.com/obophenotype/ABA_Uberon/master/src/ontology/sources/10.ofn explain --reasoner ELK  -M unsatisfiability --unsatisfiable all --explanation unsats.md --output unsat_all_explanation.ofn
```
List of unsat classes: [unsats.md](https://github.com/obophenotype/uberon/files/11573010/unsats.md)

My usecase is utilizing a subset of the human brain atlas, so I'm primarily interested in these unsatisfiable classes: `HBA:9418`, `HBA:9419`, `HBA:9710`, `HBA:9377`, `HBA:9421`, `HBA:9422`, `HBA:9708`, `HBA:9352`, `HBA:9353`

Can you help me solve these unsat classes?

**Comment by dosumis** [comment]
Prelim notes:

Lots of inconsistencies stem from general classification under [regional part of brain](http://purl.obolibrary.org/obo/UBERON_0002616) - in the main HBA file, rather than the bridge.  This causes problems for structures mapped to terms for spaces or to structures that extend outside the brain. A quick fix would be to remove this classification for all cases where we have a more specific classification in the bridge.   In the case of spaces though, we might want to consider some remapping as we almost certainly need to be referring to the tissue surrounding the space.

**Comment by hkir-dev** [comment]
HBA ontology is [automatically generated](https://github.com/obophenotype/ABA_Uberon/blob/master/src/ontology/Makefile) from the structure graph (http://api.brain-map.org/api/v2/structure_graph_download/10.json) using [linkml-owl](https://github.com/obophenotype/ABA_Uberon/blob/master/src/linkml/structure_graph_schema.yaml).

**Comment by anitacaron** [comment]
@gouttegd could you please check if there still are unsats with HBA bridge?

**Comment by gouttegd** [comment]
Yes. And I concur with @dosumis ’s preliminary notes above: Most, if not all, of the unsats seem to be caused by the fact that HBA’s terms for sulci are explicitly classified under Uberon’s [regional part of brain](http://purl.obolibrary.org/obo/UBERON_0002616), _in addition_ to being mapped to the corresponding sulcus term in Uberon.

That is, we have things like that:

* in HBA itself: ['frontal lobe sulci' (HBA:9354)](http://purl.obolibrary.org/obo/HBA_9354) SubClassOf: ['regional part of brain' (UBERON:0002616)](http://purl.obolibrary.org/obo/UBERON_0002616)
* in the Uberon-to-HBA bridge: ['frontal lobe sulci' (HBA:9354)](http://purl.obolibrary.org/obo/HBA_9354) EquivalentTo: ['frontal sulcus' (UBERON:0014639)](http://purl.obolibrary.org/obo/UBERON_0014639) and ([part of](http://purl.obolibrary.org/obo/BFO_0000050) some [Homo sapiens](http://purl.obolibrary.org/obo/NCBITaxon_9606))

This is problematic because in Uberon, a _sulcus_ is considered to be an anatomical space (“a depression or fissure in the surface of the brain”), ergo an immaterial entity, whereas a _regional part of brain_ is an anatomical structure, ergo a material entity (which may happen to contain some anatomical spaces, but cannot be an anatomical space itself).

HBA seems to use _sulcus_ to refer to the regions around the depression, rather than the depression itself. I have no strong opinion as to whether this is more correct than Uberon’s acceptation of the term (in particular, no idea how neuroscientists typically use “sulcus”), but in any case the two acceptations are clearly incompatible.

And more generally, I don’t think it is a good idea to have a term that is both (1) explicitly classified under a Uberon term and (2) mapped to another Uberon term, as is the case here. The purpose of the mapping is to create a bridging axiom between the foreign term and the Uberon term, and the purpose of that bridging axiom is to place the foreign term at a correct position within Uberon’s hierarchy when the two ontologies are merged. If the foreign term already has an explicit SubClassOf relationship with a Uberon term, then it already has a place within the hierarchy and does not need the bridging axiom.

Possible solutions:

1) Change Uberon’s classification of [sulcus](http://purl.obolibrary.org/obo/UBERON_0000093) as a [anatomical space](http://purl.obolibrary.org/obo/UBERON_0000464), so that the term makes no assertion about the “materiality” or “immateriality” of a sulcus. It’s the easiest solution as it’s basically just one axiom to remove in Uberon, but it’s also the worst in my opinion. Uberon is internally consistent around its use of “sulcus” defined as an anatomical space, I don’t like the idea of removing a useful axiom just to fix this particular problem.

2) Stop classifying HBA’s “sulcus” terms as [regional part of brain](http://purl.obolibrary.org/obo/UBERON_0002616). Assuming Uberon is correct in considering that the sulci are only the fissures and not the tissues around them, this would be the most correct solution IMHO. But then it must happen in HBA, there’s nothing we can do about that in Uberon.

3) Stop mapping HBA’s “sulcus” terms with Uberon’s “sulcus” terms. If HBA wants to keep using those terms to refer to more than the fissures, then we can just admit that this is incompatible with Uberon’s definition of a sulcus, and so the HBA terms and the Uberon terms should _not_ be mapped since they do not mean the same thing.

3a) Create new terms in Uberon to represent “the neuronal tissues that surround a sulcus” (classified as anatomical structures, not anatomical spaces), and re-map HBA’s “sulcus” terms to those new terms.

**Comment by patrick-lloyd-ray** [comment]
After discussion on our end, solution 2 is correct.

**Comment by dosumis** [comment]
> The HBA, DHBA, and PBA bridges are still locally generated and currently use “OBO” PURLs (http://purl.obolibrary.org/obo), but I can switch them to https://purl.brain-bican.org/ontology-type PURLs anytime.

That would be good.

**Comment by gouttegd** [comment]
A large part of the unsats were fixed with https://github.com/brain-bican/human_brain_atlas_ontology/pull/1, but I still count ~120 remaining unsats.

Many of them involve cranial nerves.

* [cranial nerves (HBA)](https://purl.brain-bican.org/ontology/hbao/HBA_9299) is [part of](http://purl.obolibrary.org/obo/BFO_0000050) some [myelencephalic white matter (HBA)](https://purl.brain-bican.org/ontology/hbao/HBA_9298)
* which is the human equivalent of [white matter of myelencephalon](http://purl.obolibrary.org/obo/UBERON_0019262)
* which is a [white matter of hindbrain](http://purl.obolibrary.org/obo/UBERON_0019258)
* which is a [white matter](http://purl.obolibrary.org/obo/UBERON_0002316) that is [part_of](http://purl.obolibrary.org/obo/BFO_0000050) some [hindbrain](http://purl.obolibrary.org/obo/UBERON_0002028)
* where [white matter](http://purl.obolibrary.org/obo/UBERON_0002316) is [part of](http://purl.obolibrary.org/obo/BFO_0000050) some [central nervous system](http://purl.obolibrary.org/obo/UBERON_0001017)
* ==> so ultimately [cranial nerves (HBA)](https://purl.brain-bican.org/ontology/hbao/HBA_9299) is [part of](http://purl.obolibrary.org/obo/BFO_0000050) some [central nervous system](http://purl.obolibrary.org/obo/UBERON_0001017)

But:

* [cranial nerves (HBA)](https://purl.brain-bican.org/ontology/hbao/HBA_9299) is also the human equivalent of [cranial nerve (UBERON)](http://purl.obolibrary.org/obo/UBERON_0001785)
* which is a [nerve](http://purl.obolibrary.org/obo/UBERON_0001021)
* which is [part of](http://purl.obolibrary.org/obo/BFO_0000050) some [peripheral nervous system](http://purl.obolibrary.org/obo/UBERON_0000010)

which results in [cranial nerves (HBA)](https://purl.brain-bican.org/ontology/hbao/HBA_9299) being part of both the CNS and the PNS, something that Uberon does not allow (there’s an explicit disjointness axiom between '[part of](http://purl.obolibrary.org/obo/BFO_0000050) some [central nervous system](http://purl.obolibrary.org/obo/UBERON_0001017)' and '[part of](http://purl.obolibrary.org/obo/BFO_0000050) some [peripheral nervous system](http://purl.obolibrary.org/obo/UBERON_0000010)'

I believe the faulty axiom is the first: [cranial nerves (HBA)](https://purl.brain-bican.org/ontology/hbao/HBA_9299) is [part of](http://purl.obolibrary.org/obo/BFO_0000050) some [myelencephalic white matter (HBA)](https://purl.brain-bican.org/ontology/hbao/HBA_9298). “white matter” belongs to the CNS while “nerve“ is strictly defined (in Uberon at least) as belonging to the PNS, so stating that “cranial nerves” are part of the “white matter“ is inconsistent.

**Comment by patrick-lloyd-ray** [comment]
Thanks for this, @gouttegd. I'm going to look into these and work through them.

**Comment by dosumis** [comment]
Worth searching through previous tickets.  We've definitely been here before.  One possible solution is to classify as nerve roots rather than nerves.  See https://www.ebi.ac.uk/ols4/ontologies/uberon/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FUBERON_0006843?lang=en
